src/js/modules/RoutingApp.js

```diff
@@ -25,186 +28,174 @@ import {
 } from "./core/components";
 import { DescriptionPage } from "./descriptions/components";
 import { SectionPage } from "./sections/components";
-import {
-  ContributorPage,
-  PhotographerPage,
-  IllustratorPage,
-} from "./users/components";
+import { ContributorPage, ArtistPage } from "./users/components";
+
+import { sessionify } from "./accounts/actions";
 
-import { fetchAllData } from "./core/actions";
-import { sessionfy } from "./accounts/actions";
+const RoutingAppQuery = gql`
+  query RoutingAppQuery {
+    allSections {
+      id
+      name
+      permalink
+    }
+  }
+`;
 
-class RoutingApp extends Component {
+class RoutingApp extends PureComponent {
   constructor(props) {
     super(props);
   }
 
   componentDidMount() {
-    this.prepareData();
-  }
+    const { sessionify } = this.props;
 
-  // Separate function because it is reused for the DataErrorPage
-  prepareData = () => {
-    this.props.fetchAllData();
     const session = localStorage.getItem("session");
     if (session) {
-      this.props.sessionfy(JSON.parse(session));
+      sessionify(JSON.parse(session));
+    } else {
+      const urlHeaders = queryString.parse(window.location.search);

```

Comment:
Confused. When does this situation happen? When the token, client_id, uid are in the url query parameters?

Reply:
When you click "Confirm my account" in the confirmation email devise sends, it has these parameters. It'd be nice if, once you confirm your account, you're logged in. I'll add a comment.